packages/website/components/PageContent.tsx

```diff
@@ -1,194 +1,199 @@
+      gridTemplateColumns: '1fr 7fr 2fr',
+      gridTemplateAreas: `
+        ". header header"
+        ". content toc"
+      `,
+    },
+    '@media screen and (min-width: 1700px)': {
+      gridTemplateColumns: '1fr 6fr 2fr 1fr',
+      gridTemplateAreas: `
+        ". header header ."
+        ". content toc ."
+      `,
+    },
+  }),
+  header: css({
+    gridArea: 'header',
+    display: 'grid',
+    gridTemplateColumns: '8fr 2fr',
+    gridAutoRows: 'min-content',
+    paddingTop: tokens.spacing4Xl,
+    paddingBottom: tokens.spacingM,
+    borderBottom: `1px solid ${tokens.gray300}`,
+    marginBottom: tokens.spacing2Xl,
+    '@media screen and (min-width: 1440px)': {
+      gridTemplateColumns: '7fr 2fr',
+    },
+    '@media screen and (min-width: 1700px)': {
+      gridTemplateColumns: '6fr 2fr',
+    },
+    // this selector will make sure that all children of the header will start at the first column of its grid
+    '> *': {
+      gridColumnStart: 1,
+    },
+  }),
+  content: css({
+    gridArea: 'content',
+  }),
+  article: css({
+    // this style makes sure that the first element of the content doesn't have extra spacing
+    '> *:first-child': { marginTop: 0 },
+  }),
+  tableOfContent: css({
+    gridArea: 'toc',
+    display: 'flex',
+    flexDirection: 'column',
+    position: 'sticky',
+    top: tokens.spacing2Xl,
+    paddingLeft: tokens.spacing2Xl,
+    alignSelf: 'start',
+    overflowY: 'auto',
+    overscrollBehavior: 'contain',
+  }),
 };
 
-function PageHeader(props: {
-  title?: string;
-  storybook?: string;
-  github?: string;
-  status?: string;
-}) {
+interface PageHeaderProps {
+  title: FrontMatter['title'];
+  github?: FrontMatter['github'];
+  description?: FrontMatter['description'];
+  status?: FrontMatter['status'];
+}
+
+function PageHeader({ title, github, description, status }: PageHeaderProps) {
+  const showNote = status === 'deprecated';
+
   return (
     <header className={styles.header}>
-      {props.title && (
-        <DisplayText as="h1" marginBottom="spacingL">
-          {props.title}
-        </DisplayText>
+      <Flex
+        alignItems="flex-start"
+        justifyContent="space-between"
+        marginBottom={showNote ? 'spacingXl' : 'spacingXs'}
+      >
+        {title && (
+          <DisplayText as="h1" marginBottom="none">
+            {title}
+          </DisplayText>
+        )}
+
+        {github && (
+          <TextLink
+            href={github}
+            target="_blank"
+            rel="noopener noreferrer"
+            icon={<ExternalLinkIcon />}
+            alignIcon="end"
+          >
+            View on Github
+          </TextLink>
+        )}
+      </Flex>
+
+      {showNote && (
+        <Flex flexDirection="column" marginBottom="spacingXl">
+          <Note variant="negative" title="Deprecated component">
+            {title} was deprecated in February 2022. It will be deleted from the
+            repository in 6 months.
```

Comment:
What about making it more clear?
```suggestion
            {title} was deprecated in v4. It will be deleted from the
            repository on 12th July 2022.
```

Reply:
I liked the idea

cc @domarku